file_catalog/mongo.py

```diff
@@ -93,134 +112,207 @@ def find_files(self, query={}, keys=None, limit=None, start=0):
+
+        Keyword Arguments:
+            query -- MongoDB query
+            keys -- fields to include in MongoDB projection
+            limit -- max count of files returned
+            start -- starting index
+
+        Returns:
+            List of MongoDB files
+        """
+        projection = Mongo._get_projection(
+            keys, default={"uuid": True, "logical_name": True}
+        )
+        result = self.client.files.find(query, projection)
+        ret = Mongo._limit_result_list(result, limit, start)
+
         return ret
 
     @run_on_executor
-    def create_file(self, metadata):
+    def count_files(  # pylint: disable=W0613
+        self, query: Optional[Dict[str, Any]] = None, **kwargs: Any,
+    ) -> int:
+        """Get count of files matching query."""
+        if not query:
+            query = {}
+        ret = self.client.files.count_documents(query)
+        return cast(int, ret)
+
+    @run_on_executor
+    def create_file(self, metadata: Dict[str, Any]) -> str:
+        """Insert file metadata.
+
+        Return uuid.
+        """
         result = self.client.files.insert_one(metadata)
         if (not result) or (not result.inserted_id):
-            logger.warn('did not insert file')
-            raise Exception('did not insert new file')
-        return metadata['uuid']
+            logger.warning("did not insert file")
+            raise Exception("did not insert new file")
+        return cast(str, metadata["uuid"])
 
     @run_on_executor
-    def get_file(self, filters):
-        return self.client.files.find_one(filters, {'_id':False})
+    def get_file(self, filters: Dict[str, Any]) -> Dict[str, Any]:
+        """Get file matching filters."""
+        file = self.client.files.find_one(filters, {"_id": False})
+        return cast(Dict[str, Any], file)
 
     @run_on_executor
-    def update_file(self, uuid, metadata):
-        result = self.client.files.update_one({'uuid': uuid},
-                                              {'$set': metadata})
+    def update_file(self, uuid: str, metadata: Dict[str, Any]) -> None:
+        """Update file."""
+        result = self.client.files.update_one({"uuid": uuid}, {"$set": metadata})
 
         if result.modified_count is None:
-            logger.warn('Cannot determine if document has been modified since `result.modified_count` has the value `None`. `result.matched_count` is %s' % result.matched_count)
+            logger.warning(
+                "Cannot determine if document has been modified since `result.modified_count` has the value `None`. `result.matched_count` is %s",
+                result.matched_count,
+            )
         elif result.modified_count != 1:
-            logger.warn('updated %s files with id %r',
-                        result.modified_count, uuid)
-            raise Exception('did not update')
+            logger.warning("updated %s files with id %r", result.modified_count, uuid)
+            raise Exception("did not update")
```

Comment:
`"did not update"` doesn't match so well with the logging statement above it.
Consider building a string and using it in both the logger warning and the exception message?

Reply:
sure thing